static/app/views/onboarding/targetedOnboarding/types.ts

```diff
@@ -27,3 +24,23 @@ export type StepDescriptor = {
   title: string;
   hasFooter?: boolean;
 };
+
+export type ClientState = {
+  // map from platform id to project id. Contains projects ever created by onboarding.
+  platforms: {[key in PlatformKey]?: string};
+
+  // Contains platforms currently selected. This is different from `platforms` because
+  // a project created by onboarding could be unselected by the user in the future.
+  selectedPlatforms: PlatformKey[];
+};
+
+export function fetchClientState(api: Client, orgSlug: string): Promise<ClientState> {
+  return api
+    .requestPromise(`/organizations/${orgSlug}/client-state/onboarding/`)
+    .then(lastState => {
```

Comment:
@Neo-Zhixing nit: you can make this use `async/await` instead of `.then` here

Reply:
I think .then is slightly more descriptive here. Should be the same.

Reply:
@Neo-Zhixing what do you mean "more descriptive"? I'm not seeing how that is. Anyways, we have a TSC guideline to use aysnc/await when we can: https://www.notion.so/sentry/Adopt-async-await-100-for-requests-dynamic-imports-d23d80aa5ffa4f049ba737360730eaed